.github/workflows/ci.yaml

```
@@ -0,0 +1,44 @@
+name: Continuous Integration
+
+on:
+    push:
+        branches: [master]
+    pull_request:
+        branches: [master]
+
```

Missing trigger for release?

```suggestion
    release:
        types: [created]

```

---

It's triggered automatically on every changed with `tests` passing (that's why the `need` statement). Then, `semantic-release` decides whether to release a new version or not.

---

If for some reason you prefer to control your release cycle, we should have then a new branch such as develop publishing only to GitHub Releases as pre-releases, and then when you want to release officialy, merge the changes on `master`.

---

Sounds good :), we can use `master` branch for development, and `release` for releasing.

---

https://semantic-release.gitbook.io/semantic-release/support/faq#is-it-really-a-good-idea-to-release-on-every-push